### PR TITLE
`scroll()`

### PR DESCRIPTION
**Pull request by mattgperry**
This PR adds a new universal API, `scroll()`. This function provides a scroll progress value to callbacks. It is powered by `ScrollTimeline`, where available, for 40% less per-frame JS and no DOM measurements.

The intention is to, in the future, expand this function to:
- Support element tracking via `ViewTimeline`
- Accept `animate()` animations for hardware accelerated animations

### CODE REVIEW

File: packages/framer-motion/src/render/dom/scroll/track.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import { frame, cancelFrame, frameData } from "../../../frameloop"
```
Comment (mattgperry) [inline packages/framer-motion/src/render/dom/scroll/track.ts:1]: This is the previous `scroll/index` and shouldn't need a review.

**Comment by adamseckel** [comment]
Just confirming: `scroll` now returns only a progress number, and the previous `scroll` is now `scrollInfo`?

**Comment by mattgperry** [comment]
@adamseckel Yeah that's right. Ideally this will be the main API going forward as it can migrate off of the `scrollInfo` fallback over time.

File: packages/framer-motion/src/utils/__tests__/delay.test.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ describe("delay", () => {
     test("provides overshoot duration", async () => {
         return new Promise<void>((resolve) => {
             delay((overshoot) => {
-                expect(overshoot).toBeLessThan(20)
+                expect(overshoot).toBeLessThan(50)
```
Comment (adamseckel) [inline packages/framer-motion/src/utils/__tests__/delay.test.ts:19]: Why did this change?

Comment (mattgperry) [reply packages/framer-motion/src/utils/__tests__/delay.test.ts:19]: Oh flaky test

**Review by adamseckel — APPROVED** [review]
Code LGTM (strange browser api choices aside 🙈)

What's your plan for rolling this out? It's a breaking change since the `scroll` export is now quite different. Do we need a plan to address that in Framer where there may be user code built on existing `scroll` behaviour?

**Pull request merged**